### PR TITLE
Summary reports job

### PR DESCRIPTION
**Pull request by glicht**
## Status
Ready

## Related Issues
Related: https://github.com/demisto/etc/issues/17124

## Description
Example Playbook to be used in a job to generate summary reports of a completed incidents

## Required version of Demisto
4.1.0

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation (as part of playbook)
- [ ] Code Review

## Additional
Contains update to configure tests so it will always configure a tests filter file even if there are missing tests. That way the build will continue (will still fail) but developer will be able to know if the tests that are configured if they are working. Also modified pre commit hook to check the configure tests so we give an error before commit if tests are missing. Example of a failed build that is missing tests but continues to run the tests that are configured: https://circleci.com/gh/demisto/content/24644

### CODE REVIEW

**Comment by glicht** [comment]
@kirbles19 Files for review.

* Playbook `description`: https://github.com/demisto/content/edit/summary-reports-job/Playbooks/playbook-Send_Investigation_Summary_Reports.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F3744
* Job Playbook `description`: https://github.com/demisto/content/edit/summary-reports-job/Playbooks/playbook-Send_Investigation_Summary_Reports_Job.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F3744
* Script `comment`: https://github.com/demisto/content/edit/summary-reports-job/Scripts/script-GenerateSummaryReports.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F3744

File: Tests/scripts/configure_tests.py [hunk 1]
```diff
@@ -515,16 +522,18 @@ def get_test_list(files_string, branch_name):
         tests.add("Run all tests")
 
     if sample_tests:  # Choosing 3 random tests for infrastructure testing
-        print_warning('Running sample tests due to: {}'.format(','.join(sample_tests)))
+        print_warning('Collecting sample tests due to: {}'.format(','.join(sample_tests)))
         test_ids = get_test_ids(check_nightly_status=True)
         initial_tests_len = len(tests)
+        rand = random.Random(files_string + branch_name)
```
Comment (yaakovi) [inline Tests/scripts/configure_tests.py]: why not using the default seed (time)?

File: Scripts/script-GenerateSummaryReports.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+commonfields:
```
Comment (yaakovi) [inline Scripts/script-GenerateSummaryReports.yml]: why javascript?

File: Scripts/script-GenerateSummaryReports.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  description: The report name
+  type: string
+- contextPath: SummaryReports.File
+  description: File name of the report file
+  type: string
+- contextPath: SummaryReports.FileID
+  description: File ID of the generated report
+  type: string
+scripttarget: 0
+subtype: python2
```
Comment (yaakovi) [inline Scripts/script-GenerateSummaryReports.yml]: why you subtype python if the script is javascript?

**Comment by kirbles19** [comment]
All files updated.

File: Scripts/script-GenerateSummaryReports.yml (hunk 3 above)
Comment (glicht) [reply Scripts/script-GenerateSummaryReports.yml]: pushed a commit to remove this

**Pull request merged**